example/csv/src/main/java/org/apache/calcite/adapter/csv/CsvTableFactory.java

```diff
@@ -43,8 +44,8 @@ public CsvTableFactory() {
   public CsvTable create(SchemaPlus schema, String name,
       Map<String, Object> operand, RelDataType rowType) {
     String fileName = (String) operand.get("file");
-    final File base =
-        (File) operand.get(ModelHandler.ExtraOperand.BASE_DIRECTORY.camelName);
+    final URI uri = (URI) operand.get(ModelHandler.ExtraOperand.BASE_DIRECTORY.camelName);
```

Comment:
@walterddr , this looks like a backward-incompatible change. In other words, clients have to update their code even though they don't use URIs yet.

Have you though of a possibility to keep previous code intact while allowing to leverage new API if required?

Reply:
hmm. this was confusing to me. I was not sure about the original implementation to directly convert the `Object` to `File`. 

How about:
```
Object baseDirObj = operand.get(ModelHandler.ExtraOperand.BASE_DIRECTORY.camelName);
Final File base;
if (baseDirObj instanceof URI) {
  uri = (URI) baseDirObj;
  base = new File(uri);
} else if (baseDirObj instanceof File) {
  base = (File) baseDirObj;
} else {
  throw IllegalArgumentException(...);
}
```